datahub-web-react/src/app/ingestV2/executions/components/ExecutionDetailsModal.tsx

```diff
@@ -170,82 +175,176 @@ export const ExecutionDetailsModal = ({ urn, open, onClose }: Props) => {
-                                The configurations used for this sync with the data source.
-                            </SubHeaderParagraph>
-                        </SectionSubHeader>
-                        <DetailsContainer
-                            areDetailsExpandable={isRecipeExpandable}
-                            showExpandedDetails={showExpandedRecipe}
-                        >
-                            <Typography.Paragraph ellipsis>
-                                <pre>{`${recipe}${!showExpandedRecipe && isRecipeExpandable ? '...' : ''}`}</pre>
-                            </Typography.Paragraph>
-                        </DetailsContainer>
-                        {isRecipeExpandable && (
-                            <ShowMoreButton type="link" onClick={() => setShowExpandedRecipe((v) => !v)}>
-                                {showExpandedRecipe ? 'Hide' : 'Show More'}
-                            </ShowMoreButton>
-                        )}
-                    </RecipeSection>
-                )}
-            </Section>
+                </RecipeSection>
+            )}
+        </Section>
+    );
+};
+
+const LogsTab = ({ urn, data }: { urn: string; data: GetIngestionExecutionRequestQuery | undefined }) => {
+    const output = data?.executionRequest?.result?.report || 'No output found.';
+
+    const downloadLogs = () => {
+        downloadFile(output, `exec-${urn}.log`);
+    };
+
+    return (
+        <LogsSection>
+            <SectionHeader level={5}>Logs</SectionHeader>
+            <SectionSubHeader>
+                <SubHeaderParagraph type="secondary">View logs that were collected during the sync.</SubHeaderParagraph>
+                <Button type="text" onClick={downloadLogs}>
+                    <DownloadOutlined />
+                    Download
+                </Button>
+            </SectionSubHeader>
+            <Typography.Paragraph ellipsis>
+                <pre>{output}</pre>
+            </Typography.Paragraph>
+        </LogsSection>
+    );
+};
+
+const RecipeTab = ({ data }: { data: GetIngestionExecutionRequestQuery | undefined }) => {
+    const recipeJson = data?.executionRequest?.input?.arguments?.find((arg) => arg.key === 'recipe')?.value;
+    let recipeYaml: string;
+    try {
+        recipeYaml = recipeJson && YAML.stringify(JSON.parse(recipeJson), 8, 2).trim();
+    } catch (e) {
+        recipeYaml = '';
+    }
+
+    return (
+        <RecipeSection>
+            <SectionHeader level={5}>Recipe</SectionHeader>
+            <SectionSubHeader>
+                <SubHeaderParagraph type="secondary">
+                    The configurations used for this sync with the data source.
+                </SubHeaderParagraph>
+            </SectionSubHeader>
+            <Typography.Paragraph ellipsis>
+                <pre>{recipeYaml || 'No recipe found.'}</pre>
+            </Typography.Paragraph>
+        </RecipeSection>
+    );
+};
```

Comment:
i'd recommend putting each component into its own file for readability before putting up a final pr

Reply:
Done